packages/block-library/src/legacy-widget/edit/widget-type-selector.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { Spinner, SelectControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+export default function WidgetTypeSelector( { selectedId, onSelect } ) {
+	const widgetTypes = useSelect( ( select ) => {
+		const hiddenIds =
+			select( blockEditorStore ).getSettings()
+				?.widgetTypesToHideFromLegacyWidgetBlock ?? [];
+		return select( coreStore )
+			.getWidgetTypes( { per_page: -1 } )
+			?.filter( ( widgetType ) => ! hiddenIds.includes( widgetType.id ) );
+	}, [] );
+
+	if ( ! widgetTypes ) {
+		return <Spinner />;
+	}
+
+	if ( widgetTypes?.length === 0 ) {
```

Comment:
I think there's no need to add the optional chaining here.
```suggestion
	if ( widgetType.length === 0 ) {
```